lib/rules/method.js

```diff
@@ -122,6 +122,7 @@ function checkCallExpression(ruleHelper, callExpr, node) {
     case "ThisExpression":
     case "NewExpression":
     case "TSTypeAssertion":
+    case "TypeCastExpression":
```

Comment:
In hindsight, I think this is incorrect and we need to ensure the typecast will be entered..

This ought to be tested with cases like so
*  `node.(insertAdjacentHTML: function)('afterend', potentitallybad)` (disallowed)
*  `node.(insertAdjacentHTML: function)('afterend', 'static string')` (allowed).
*  `(coolFunction: function)('afterend', potentiallybad)` (allowed).

Reply:
@mozfreddyb, thanks for taking a look again! For those first two cases (a typecast while calling the `insertAdjacentHTML` function), that appears to be invalid syntax. I get an unexpected token from the babel eslint parser, and confirmed in the [flow playground](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgBycAJgQLxgDeqYYAlgHYDOeAThgIIkBWAhgGM8jDAAkAKgFkAMgC4wACmYY2TAOYAaMMtWM1ASjDkAfGABuceiQ2oAvuhh4MYRqQLziZI9VsBudAquZB5u+gB0TKwc3PxCIhIyCgDkAEZ4UHBseGlqTElaSWx8SDrqSfr+qAJwLM6R7Fy8gsJiUtLeChlw8qV6Wil8bD0q6oYmPv4K9dFNca0y8lAArowCGPQ1+sl8UBjswiT5YEnKfOsC2iN65ZVBeGFTLA0xzfFtiytrG4xbSTt7WUYhwKp3Ol10ahuQA). It looks like you can cast the node and function parameters, but can't cast a function on a node while calling it. 

Based on that I think ignoring that expression there is OK, and in line with what was needed of TypeScript. I did, however, notice some holes in my testing so I added more cases to verify.

Does that make sense? I'm happy to make other changes if you have suggestions.

Reply:
Update: I did find one case that isn't covered by the current implementation. Looking into it!

```
(insertAdjacentHTML: function)('beforebegin', unsafe);
```

Reply:
Fixed, see my new comment!